modules/imgcodecs/src/grfmt_avif.cpp

```diff
@@ -148,11 +148,14 @@ AvifDecoder::~AvifDecoder() {
 size_t AvifDecoder::signatureLength() const { return kAvifSignatureSize; }
 
 bool AvifDecoder::checkSignature(const String &signature) const {
-  avifDecoderSetIOMemory(decoder_,
+  avifDecoder *decoder = avifDecoderCreate();
+  if (!decoder) return false;
+  avifDecoderSetIOMemory(decoder,
                          reinterpret_cast<const uint8_t *>(signature.c_str()),
                          signature.size());
-  decoder_->io->sizeHint = 1e9;
-  const avifResult status = avifDecoderParse(decoder_);
+  decoder->io->sizeHint = 1e9;
```

Comment:
The sizeHint is just part of the internal logic. AVIF is a format that does not have a signature, and you could have valid AVIF's that would require to parse a lot before being sure. Still, with 500 bytes, as defined at https://github.com/opencv/opencv/blob/fdc0c12b7fff00e1b41f75e11401073eee23b7e7/modules/imgcodecs/src/grfmt_avif.cpp#L136 , you should be able to detect most AVIFs as built with common tools.

Reply:
Ok, let's reduce hint then, or may be do not use it.

Reply:
We can set it to 0 and less checks are done: https://github.com/AOMediaCodec/libavif/blob/d70b3baa791ba5b010b931e3970aabba299a7e36/include/avif/avif.h#L780
We actually should do that as otherwise we need to try reading the whole file.

Reply:
Done